hasher-matcher-actioner/hmalib/lambdas/api/action_rules_api.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import bottle
+import typing as t
+import sys
+
+from botocore.exceptions import ClientError
+from bottle import response
+from dataclasses import dataclass, field
+from hmalib.common.logging import get_logger
+from .middleware import jsoninator, JSONifiable, DictParseable
+from hmalib.common.config import HMAConfig
+from hmalib.common import config as hmaconfig
+from hmalib.common.evaluator_models import ActionRule
+from hmalib.common.classification_models import Label, ActionLabel
+
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class ActionRulesAPIRequest(DictParseable):
+    action_rule: ActionRule
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "ActionRulesAPIRequest":
+        ar = ActionRule.from_aws(d["action_rule"])
+        logger.info("Deserialized ActionRule: %s", ar)
+        return cls(ar)
+
+
+@dataclass
+class ActionRulesAPIResponse(JSONifiable):
+    error_message: str
+    action_rules: t.List[ActionRule] = field(default_factory=list)
+
+    def to_json(self) -> t.Dict:
+        return {
+            "error_message": self.error_message,
+            "action_rules": [action_rule.to_aws() for action_rule in self.action_rules],
+        }
+
+
+def handle_unexpected_error(e: Exception):
```

Comment:
I like this, I wonder if we want it as a top level decorator for all endpoints...

Reply:
I agree. I think when we're done with V0 a thing to come back and standardize is API response consistency, including error handling. Added a TODO and filed Issue 599: https://github.com/facebook/ThreatExchange/issues/599